### DISCUSSION TITLE
Sidekiq stops processing queue after 3.5.0 update

### DISCUSSION THREAD

**Discussion by verymilan** [mastodon/mastodon#17932] [Q&A]
Hi there, i started having issues with the queue after the recent update and i am not sure if the problem lays on my end (eg. configuration mistake).

Sidekiq logs a number of 
```
WARN: HTTP::ConnectionError: failed to connect: Too many open files @ rb_sysopen - /etc/resolv.conf on https://domain.xyz/inbox
WARN: /.../.rbenv/versions/3.0.3/lib/ruby/3.0.0/resolv.rb:961:in `initialize'
```
exceptions while puma shows a number of "too many requests" for the OVH object storage around that time. I have noticed the hint about S3 single request in the update notes, but would it have effect for the SWIFT module?

I am not sure how to read the sidekiq error bo be honest. It sounds unusual and confusing for me. I mean i can raise the hard/soft-limit, but isn't the problem laying somewhere else here? :thinking:
****
Update: according to users, there are more instances with similar symptoms (timeline not properly updating in particular). My current workaround is a restart cron for every 3h.

**Comment by ClearlyClaire** [comment]
This is a file descriptor leak, and considering I haven't seen any report of it yet, nor have I encountered it myself, it is likely to be configuration-dependent. It could be in the OpenStack SWIFT handling code indeed, since it's relatively infrequently used.

It may help if you could post the output of something like `lsof -p PID` where `PID` is the process ID of one of the `sidekiq` processes.

**Reply by todoneunl** [reply]
I encountered this too. The open files was set to the default of 1024 and Sidekiq had 1025 open. I increased the open files limit permanently to a much higher number for all accounts on my VPS (40000). 

Edit `/etc/security/limits.conf` and add somewhere (e.g. at the bottom) `* - nofile 40000`. Restart Sidekiq using `systemctl restart mastodon-sidekiq`. More info about increasing file limits here: https://tecadmin.net/increase-open-files-limit-ubuntu/ (I didn't had to enable that PAM module, but maybe that's because I use Debian)

**Comment by weingaertnerit** [comment]
Same Issue here. 
Fixed through editing sidekiq.service and adding under `[Service]`:
`LimitNOFILE=64000`

`systemctl daemon-reload` and restart sidekiq

**Reply by todoneunl** [reply]
The original problem, the reason of 'too many open files', was a very high amount of Sidekiq jobs. Today that amount came back, but without the too many open files issue (afaik). I now have split the Sidekiq process in multiple processes, like my other server (didn't have the problem). Let's see if this fix is more permanent.

**Reply by Isaw-w** [reply]
Hi, Not sure how to  split the Sidekiq process, is there any tutorial? sorry for the silly question.
